cfme/utils/appliance/__init__.py

```diff
@@ -2154,25 +2179,16 @@ def set_rubyrep_replication(self, host, port=5432, database='vmdb_production',
                                 username='root', password=None):
         """Sets up rubyrep replication via advanced configuration settings yaml."""
         password = password or self._encrypt_string(conf.credentials['ssh']['password'])
-        yaml = self.get_yaml_config()
-        if 'replication_worker' in yaml['workers']['worker_base']:
-            dest = yaml['workers']['worker_base']['replication_worker']['replication'][
-                'destination']
-            dest['database'] = database
-            dest['username'] = username
-            dest['password'] = password
-            dest['port'] = port
-            dest['host'] = host
-        else:  # 5.5 configuration:
-            dest = yaml['workers']['worker_base'][':replication_worker'][':replication'][
-                ':destination']
-            dest[':database'] = database
-            dest[':username'] = username
-            dest[':password'] = password
-            dest[':port'] = port
-            dest[':host'] = host
-            logger.debug('Dest: {}'.format(dest))
-        self.set_yaml_config(yaml)
+        dest = {'workers': {'worker_base': {'replication_worker': {'replication': {
```

Comment:
Before `dest` was the content of `yaml['workers']['worker_base']['replication_worker']['replication']['destination']`, and then 'database', 'username', 'password', 'port', and 'host' were set in that level of the dict.

Here `dest` is the dictionary containing what `dest` was before, and the statements below will set 'database' etc at the top level of `dest`.

Before: 
```python
self.set_yaml_config({'workers': {'worker_base': {'replication_worker': {'repliaction': {'destination': {
    'database': database,
    'username': username,
    'password': password,
    'port': port,
    'host': host}}}}}})
```

Now:
```python
self.set_yaml_config({{'workers': {'worker_base': {'replication_worker': {'repliaction': {'destination': {}}}}},
    'database': database,
    'username': username,
    'password': password,
    'port': port,
    'host': host})
```